src/nublado/controller/services/builder/lab.py

```diff
@@ -108,6 +110,31 @@ def __init__(
         self._discovery = discovery_client
         self._logger = logger
         self._volume_builder = VolumeBuilder()
+        self._introspect_container()
+
+    def _introspect_container(self) -> None:
+        """Determine the registry/repository/tag of the running container.
+        This comes from the Phalanx deployment chart.
+        It would be nice if the DownwardAPI gave us this, but it does not.
+        """
+        repository = os.getenv(
+            "NUBLADO_CONTROLLER_REPOSITORY", "ghcr.io/lsst-sqre/nublado"
+        )
+        # We're guessing that version 11 will be the first place this shows up.
+        tag = os.getenv("NUBLADO_CONTROLLER_TAG", "11.0.0")
```

Comment:
Would we want to know if these are not set? Is there any danger in silently defaulting to a version here? Perhaps a warning if we do want to default but want to know somehow? Otherwise if you think not then this would be fine as it is.